src/code_oscar.jl

```diff
@@ -0,0 +1,566 @@
+
+# the necessary Julia packages
+import Oscar
+import Singular
+import Polymake
+import Nemo
+import GAP
+
+# for using LaTex in docstrings processed by Documenter.jl
+import Markdown
+
+
+#############################################################################
+
+import Oscar.base_ring
+
+Oscar.base_ring(f::Oscar.MPolyElem_dec{T}) where {T} = base_ring(f.f)
```

Comment:
Please make a pull request adding this to Oscar.jl

Reply:
Yes.
(There are a few more omissions which I noticed but which eventually were not needed.)